src/Application.php

```diff
@@ -217,195 +226,47 @@ public function getAdapterFactory()
-            if (null === $this->config) {
-                $this->config = Factory::createConfig();
-            }
-
-            if ('core:init' !== $commandName) {
-                if (null === $this->adapter) {
-                    if (null === $adapter = $this->config->get('adapter')) {
-                        $adapter = $this->determineAdapter();
-                    }
-
-                    $this->buildAdapter($adapter);
-                }
-
-                if (null === $this->issueTracker) {
-                    $issueTracker = $this->config->get('issue_tracker');
-
-                    if ($issueTracker) {
-                        $this->buildIssueTracker($issueTracker);
-                    } elseif ($this->adapter instanceof IssueTracker) {
-                        $this->issueTracker = $this->adapter;
-                    } else {
-                        $message =
-                            'Adapter "%s" doesn\'t support issue-tracking and no issue tracker is configured. '.
-                            PHP_EOL.
-                            'Please run the "init" or "core:configure" command to configure a (default) issue '.
-                            'tracker.';
-
-                        throw new \RuntimeException(sprintf($message, get_class($this->adapter)));
-                    }
-                }
-            }
-        }
-
-        /** @var \Symfony\Component\Console\Helper\Helper[] $helperSet */
+        /** @var \Symfony\Component\Console\Helper\HelperSet $helperSet */
         $helperSet = $command->getHelperSet();
+
         foreach ($helperSet as $helper) {
             if ($helper instanceof OutputAwareInterface) {
                 $helper->setOutput($output);
             }
-        }
-
-        parent::doRunCommand($command, $input, $output);
-    }
-
-    /**
-     * @return Adapter
-     */
-    private function determineAdapter()
-    {
-        $builder = new ProcessBuilder(['git', 'config', '--get', 'remote.origin.url']);
-        $builder
-            ->setWorkingDirectory(getcwd())
-            ->setTimeout(3600)
-        ;
-
-        $process = $builder->getProcess();
-        $process->run();
-
-        if (!$process->isSuccessful()) {
-            throw new \RuntimeException(
-                'The adapter type could not be determined (no Git origin configured for this repository). '.
-                'Please run the "core:init" command.'
-            );
-        }
-
-        $remoteUrl = strtolower($process->getOutput());
-        $ignoredAdapters = [];
 
-        foreach (array_keys($this->getAdapterFactory()->all()) as $adapterName => $adapterInfo) {
-            $config = $this->config->get(sprintf('[adapters][%s]', $adapterName));
-
-            // Adapter is not configured ignore
-            if (null === $config) {
-                $ignoredAdapters[] = $adapterName;
-
-                continue;
+            if ($helper instanceof InputAwareInterface) {
```

Comment:
i wonder if this can be a oneliner with a ternary operator or if we are up for those things